src/Features/CSharp/Portable/Completion/CompletionProviders/NamedParameterCompletionProvider.cs

```diff
@@ -256,10 +256,9 @@ int IEqualityComparer<IParameterSymbol>.GetHashCode(IParameterSymbol obj)
         {
             return Task.FromResult<TextChange?>(new TextChange(
                 selectedItem.Span,
-                // Do not insert colon on <Tab> so that user can complete out a variable name that does not currently exist.
-                // ch == null is to support the old completion only.
-                // Do not insert an extra colon if colon has been explicitly typed.
-                (ch == null || ch == '\t' || ch == ':') ? selectedItem.DisplayText : selectedItem.GetEntireDisplayText()));
+                // Insert extra colon if commiting with '(' only: "method(parameter:(" is preferred to "method(parameter(".
+                // In all other cases, do not add extra colon. Note that colon is already added if commiting with ':'.
+                ch == '(' ? selectedItem.GetEntireDisplayText() : selectedItem.DisplayText));
```

Comment:
was this a speculative case you were worried about?  I'm just curious why you enumerated this exact specific case?

Note: i'm in general approval of the change here.  I'm just curious about the thought process that led to checking exactly for `(`.  Thanks!

Reply:
Here is it: https://github.com/dotnet/roslyn/issues/30097

Type `M(M(M(M(M`
With context of `Method(int m)`

We alternate the method and the parameter. The idea is to add colons after the parameter _assuming_ that `m:(` is more expected that `m(`. The assumption seems to be speculative: I'm ready to reject it or to add more scenarios, e.g. when committing with `!` or `~`